flink-runtime/src/main/java/org/apache/flink/runtime/jobgraph/JobVertex.java

```diff
@@ -125,6 +132,8 @@ public JobVertex(String name) {
 	public JobVertex(String name, JobVertexID id) {
 		this.name = name == null ? DEFAULT_NAME : name;
 		this.id = id == null ? new JobVertexID() : id;
+		this.operatorIDs.add(new OperatorID(this.id.getLowerPart(), this.id.getUpperPart()));
+		this.operatorIdsAlternatives.add(null);
```

Comment:
There's an implicit contract that the length or `operatorIDs` must be equivalent to `operatorIdsAlternatives`. We could store them as a pair instead.